test/test_backbone_utils.py

```diff
@@ -23,3 +25,50 @@ def test_resnet50_fpn_backbone(self):
         resnet50_fpn = resnet_fpn_backbone(backbone_name='resnet50', pretrained=False)
         y = resnet50_fpn(x)
         self.assertEqual(list(y.keys()), ['0', '1', '2', '3', 'pool'])
+
+
+class IntermediateLayerGetterTester(unittest.TestCase):
+    def test_old_new_match(self):
+        model = torchvision.models.resnet18(pretrained=False)
+
+        return_layers = {'layer2': '5', 'layer4': 'pool'}
```

Comment:
FYI, it fails when we include the final output in the return layers:
```suggestion
        return_layers = {'layer2': '5', 'layer4': 'pool', 'fc': 'fc1'}
```

with:
```
E       RuntimeError: mat1 and mat2 shapes cannot be multiplied (1024x1 and 512x1000)
```

Reply:
Thanks for the catch! I need to check more carefully, but the old implementation doesn't work in this case because of the `torch.flatten` call (which is not a `nn.Module`), but I believe this should work with the new implementation. To be verified